lib/vty.c

```diff
@@ -2227,6 +2250,7 @@ static int vtysh_flush(struct vty *vty)
 		vty_close(vty);
 		return -1;
 	case BUFFER_EMPTY:
+		vty->vty_buf_size_written = 0;
```

Comment:
if the buffer is empty, should we cancel a "vtysh_write" event?

Reply:
Correct me if I understood your question right. You are suggesting we should cancel the vtysh_write event when the buffer is empty?
In the existing code vtysh_flush is called at the end of command execution (considering outputs less than 128KB also). After that vty_close is called which will cancel the vtysh_write event since it is end of command execution. I thought it is better to leave the existing code as is and did not include intermediate cancel and starting of vtysh_write event.